src/xenia/kernel/kernel_state.cc

```diff
@@ -107,8 +111,13 @@ XModule* KernelState::GetModule(const char* name) {
     // Some games request this, for some reason. wtf.
     return nullptr;
   } else {
-    // TODO(benvanik): support user modules/loading/etc.
-    assert_always();
+    for (XUserModule *module : user_modules_) {
```

Comment:
add a std::lock_guard<std::mutex> lock(object_mutex_); -- (just in this else is fine)